File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangeBidHouseBuyResultMessage.scala





















// Generated on 05/03/2013 10:34:53
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangeBidHouseBuyResultMessage extends MessageDeserializer {
  val messageId = 6272

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ExchangeBidHouseBuyResultMessage.messageId

    val uid: Int
    val bought: Boolean

    def serialize(buf: Buffer) {
      buf.writeInt(uid)
      buf.writeBoolean(bought)
    }

  }

  protected def create(uid: Int, bought: Boolean): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val uid = buf.readInt()
    val bought = buf.readBoolean()

    create(uid, bought)
  }

}

object ExchangeBidHouseBuyResultMessage extends ExchangeBidHouseBuyResultMessage {
  case class MessageImpl(uid: Int, bought: Boolean) extends Message
  type Target = MessageImpl

  override def create(uid: Int, bought: Boolean) = MessageImpl(uid, bought)

  def apply(uid: Int, bought: Boolean) = create(uid, bought)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangeBidHouseBuyResultMessage.java


// Generated on 05/08/2013 19:37:53
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangeBidHouseBuyResultMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6272;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int uid;
    public boolean bought;
    
    public ExchangeBidHouseBuyResultMessage() { }
    
    public ExchangeBidHouseBuyResultMessage(int uid, boolean bought) {
        this.uid = uid;
        this.bought = bought;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(uid);
        buf.writeBoolean(bought);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        uid = buf.readInt();
        if (uid < 0)
            throw new RuntimeException("Forbidden value on uid = " + uid + ", it doesn't respect the following condition : uid < 0");
        bought = buf.readBoolean();
    }
    
}
